src/pudl_archiver/archivers/eia/eiarecs.py

```diff
@@ -0,0 +1,280 @@
+                data_paths_in_archive.append(output_filename)
+            download_path.unlink()
+        return data_paths_in_archive
+
+    async def __get_tab_links(self, tab_info: TabInfo, zip_path: Path) -> list[str]:
+        """Get the data files for a single tab."""
+        soup = await self.__get_soup(tab_info.url)
+        links_in_tab = soup.select("div.tab-contentbox a[href]")
+        log_scope = f"{tab_info.year}:{tab_info.name}"
+        self.logger.info(f"{log_scope}: Found {len(links_in_tab)} links")
+
+        links_filtered = [
+            link
+            for link in links_in_tab
+            if not (
+                "mailto" in link["href"].lower() or "all tables" in link.text.lower()
+            )
+        ]
+
+        self.logger.info(f"{log_scope}: Found {len(links_filtered)} relevant links")
+
+        resolved_links = [
+            urljoin(tab_info.url, link["href"]) for link in links_filtered
+        ]
+        links_with_filenames = {
+            link: f"eia-recs-{tab_info.year}-{tab_info.name}-{self.__get_filename_from_link(link)}"
+            for link in resolved_links
+        }
+
+        data_paths = await self.__add_links_to_archive(
+            links_with_filenames, zip_path=zip_path
+        )
+
+        self.logger.info(
+            f"{log_scope}: Added {len(links_with_filenames)} links to archive"
+        )
+
+        return data_paths
+
+    async def __get_tab_html_and_links(
+        self, tab_info: TabInfo, zip_path: Path
+    ) -> list[str]:
+        """Get the data files in the tab, *and* get the tab content itself."""
+        log_scope = f"{tab_info.year}:{tab_info.name}"
+        self.logger.info(f"{log_scope}: Getting links in tab")
+        links = await self.__get_tab_links(tab_info=tab_info, zip_path=zip_path)
+        self.logger.info(f"{log_scope}: Got {len(links)} links")
+
+        soup = await self.__get_soup(tab_info.url)
+        tab_content = soup.select_one("div.tab-contentbox")
+        self.logger.info(f"{log_scope}: Got {len(tab_content)} bytes of tab content")
+        html = soup.new_tag("html")
+        body = soup.new_tag("body")
+        html.append(body)
+        body.append(tab_content)
+        # TODO 2025-02-03: consider using some sort of html-to-pdf converter here.
+        # use html-sanitizer or something before feeding it into pdf.
+
+        filename = f"eia-recs-{tab_info.year}-{tab_info.name}-tab-contents.html"
+        self.add_to_archive(
+            zip_path=zip_path,
+            filename=filename,
+            blob=BytesIO(html.prettify().encode("utf-8")),
+        )
+        self.logger.info(f"{log_scope}: Added html to {zip_path} under {filename}")
+        return links + [filename]
+
+    async def __get_original_forms(self, year: int, zip_path: Path) -> list[str]:
+        """Get the survey forms that were used to collect the data.
+
+        These are all on the same page, which is different from the yearly RECS
```

Comment:
Amusingly, the 2015 forms have some that are marked 'eia-475' instead of 'eia-457'. But I checked and they seem to be a survey of residential energy consumption.